tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/Languages/DotNetLanguageSpecificChecks.cs

```diff
@@ -50,28 +45,15 @@ public async Task<CLICheckResponse> CheckGeneratedCode(string packagePath, bool
                 return new CLICheckResponse(1, "", "Failed to determine service directory from the provided package path.");
             }
 
-            var repoRoot = _gitHelper.DiscoverRepoRoot(packagePath);
-            var scriptPath = Path.Combine(repoRoot, "eng", "scripts", "CodeChecks.ps1");
-            if (!File.Exists(scriptPath))
-            {
-                _logger.LogError("Code checks script not found at: {ScriptPath}", scriptPath);
-                return new CLICheckResponse(1, "", $"Code checks script not found at: {scriptPath}");
-            }
-
-            var args = new[] { scriptPath, "-ServiceDirectory", serviceDirectory, "-SpellCheckPublicApiSurface" };
-            var options = new PowershellOptions(scriptPath, args, workingDirectory: repoRoot, timeout: CodeChecksTimeout);
-            var result = await _powershellHelper.Run(options, ct);
+            var (invoked, result) = await _repositoryScriptService.TryInvoke("CodeChecks", packagePath, new()
```

Comment:
Should `HasImplementation` be checked here in case there is an existing implementation that needs to be used instead of calling the script? I am not sure how we would use `HasImplementation`.

Reply:
`HasImplementation` just means whether there's an override script, not a fallback implementation in code. Perhaps I should rename or remove. Originally I envisioned the flow like:

```
if (repoScriptService.HasImplementation)
{
    repoScriptService.Invoke(...);
}
else
{
    fallback();
}
```

But I decided `TryInvoke` was just simpler (it checks the override + script existence under the hood). I kept `HasImplementation` in case we would want to use it anywhere, since `GetCommand` throws if the script doesn't exist on disk, but maybe it's better to just remove it?

Reply:
Yes, I think it's safe to remove it! 
I had the same thought when I read `TryInvoke` but just wanted to get it clarified.